swarm/network/kademlia_test.go

```diff
@@ -89,65 +94,209 @@ func TestNeighbourhoodDepth(t *testing.T) {
 	if depth != 0 {
-		t.Fatalf("expected depth 0, was %d", depth)
+		t.Fatalf("%d expected depth 0, was %d", testNum, depth)
 	}
+	testNum++
 
-	midAddress := pot.RandomAddressAt(baseAddress, 4)
-	midPeer := newTestDiscoveryPeer(midAddress, kad)
-	kad.On(midPeer)
+	// add a second on 7
+	kad.On(sevenPeers[1])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 5 {
-		t.Fatalf("expected depth 5, was %d", depth)
+	if depth != 0 {
+		t.Fatalf("%d expected depth 0, was %d", testNum, depth)
 	}
+	testNum++
 
-	kad.Off(midPeer)
-	depth = kad.NeighbourhoodDepth()
-	if depth != 0 {
-		t.Fatalf("expected depth 0, was %d", depth)
+	// add from 0 to 6
+	for i, p := range peers {
+		kad.On(p)
+		depth = kad.NeighbourhoodDepth()
+		if depth != i+1 {
+			t.Fatalf("%d.%d expected depth %d, was %d", i+1, testNum, i, depth)
+		}
 	}
+	testNum++
 
-	fartherAddress := pot.RandomAddressAt(baseAddress, 1)
-	fartherPeer := newTestDiscoveryPeer(fartherAddress, kad)
-	kad.On(fartherPeer)
+	kad.Off(sevenPeers[1])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 2 {
-		t.Fatalf("expected depth 2, was %d", depth)
+	if depth != 6 {
+		t.Fatalf("%d expected depth 6, was %d", testNum, depth)
 	}
+	testNum++
 
-	midSameAddress := pot.RandomAddressAt(baseAddress, 4)
-	midSamePeer := newTestDiscoveryPeer(midSameAddress, kad)
-	kad.Off(closerPeer)
-	kad.On(midPeer)
-	kad.On(midSamePeer)
+	kad.Off(peers[4])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 2 {
-		t.Fatalf("expected depth 2, was %d", depth)
+	if depth != 4 {
+		t.Fatalf("%d expected depth 4, was %d", testNum, depth)
 	}
+	testNum++
 
-	kad.Off(fartherPeer)
-	log.Trace(kad.string())
-	time.Sleep(time.Millisecond)
+	kad.Off(peers[3])
 	depth = kad.NeighbourhoodDepth()
-	if depth != 0 {
-		t.Fatalf("expected depth 0, was %d", depth)
+	if depth != 3 {
+		t.Fatalf("%d expected depth 3, was %d", testNum, depth)
+	}
+	testNum++
+}
+
+func TestHealthSimple(t *testing.T) {
+
+	// base address is all zeros
+	// no peers
+	// unhealthy (and lonely)
+	k := newTestKademlia("11111111")
+	if err := assertHealthSimple(t, k, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// know one peer but not connected
+	// unhealthy
+	Register(k, "11100000")
+	log.Trace(k.String())
+	if err := assertHealthSimple(t, k, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// know one peer and connected
+	// healthy
+	On(k, "11100000")
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+
+	// know two peers, only one connected
+	// unhealthy
+	Register(k, "11111100")
+	log.Trace(k.String())
+	if err := assertHealthSimple(t, k, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// know two peers and connected to both
+	// healthy
+	On(k, "11111100")
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+
+	// know three peers, connected to the two deepest
+	// healthy
+	Register(k, "00000000")
+	log.Trace(k.String())
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
```

Comment:
Since you pass `testing.T` to the assert function, you can just call `t.Fatal` inside and refactor all test code, to have an assert on one line, and not three, when you no longer have to return the error.

Reply:
I prefer to know the line it failed :)

Reply:
`t.Helper()` will take care of that